### PR TITLE
Move redis credentials to their own file

### PR DESCRIPTION
**Pull request by fayez-nazzal**
Move all redis credentials to their own file so that they are not repeated in each file. As credentials are used differently inside each file, added `pickCredentials` helper function to obtain a subset of the whole credentials.

### CODE REVIEW

File: server/redis-credentials.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import env from "./env";
+import pickBy from "lodash.pickby";
+
+const credentials = {
+  host: env.REDIS_HOST,
+  port: env.REDIS_PORT,
+  db: env.REDIS_DB,
+  ...(env.REDIS_PASSWORD && { password: env.REDIS_PASSWORD })
```
Comment (fayez-nazzal) [inline server/redis-credentials.ts:7]: I think not needed, just add like rest of fields and pickCredentials will filter it out if it didn't exist

Comment (hammady) [reply server/redis-credentials.ts:7]: I don't think we need to filter credentials. I know you aimed at exact behaviour but db is a standard redis configuration which defaults to 0. Let's simplify the code by requiring all redis config with no filters.

File: package-lock.json
Comment (hammady) [inline package-lock.json:1]: Are changes to this file required for this PR, especially after removing the `lodash.pickby` package?

**Review by hammady — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (fayez-nazzal) [reply package-lock.json:1]: @hammady Do you mean remove the pick logic and always consume the full credentials object?

Comment (hammady) [reply package-lock.json:1]: Yes, exactly!

**Review by hammady — APPROVED** [review]

**Pull request merged**